include/__generator.hpp

```diff
@@ -227,31 +227,45 @@ class __manual_lifetime<_T&&> {
     _T* __value_;
 };
 
+struct use_allocator_arg {};
+
 namespace ranges {
 
-template <typename _Rng>
+template <typename _Rng, typename _Allocator = use_allocator_arg>
 struct elements_of {
     explicit constexpr elements_of(_Rng&& __rng) noexcept
-    : __range_(static_cast<_Rng&&>(__rng))
+    requires std::is_default_constructible_v<_Allocator>
+    : __range(static_cast<_Rng&&>(__rng))
     {}
 
+    constexpr elements_of(_Rng&& __rng, _Allocator&& __alloc) noexcept
+    : __range((_Rng&&)__rng), __alloc((_Allocator&&)__alloc) {}
+
     constexpr elements_of(elements_of&&) noexcept = default;
 
     constexpr elements_of(const elements_of &) = delete;
     constexpr elements_of &operator=(const elements_of &) = delete;
     constexpr elements_of &operator=(elements_of &&) = delete;
 
-    constexpr _Rng&& get() && noexcept {
-        return static_cast<_Rng&&>(__range_);
+    constexpr _Rng&& get() noexcept {
+        return static_cast<_Rng&&>(__range);
+    }
+
+    constexpr _Allocator&& get_allocator() noexcept {
+        return static_cast<_Allocator&&>(__alloc);
     }
 
 private:
-    _Rng &&__range_; // \expos
+    [[no_unique_address]] _Allocator __alloc; // \expos
```

Comment:
Why is this taking a copy of the allocator if passed an rvalue but passing through a reference if passed an lvalue?
This seems inconsistent with the behaviour of the __range_ parameter which is always passed through by reference.

Can we add a comment here indicating why this is the case?